pkg/steps/secretrecordingclient/client.go

```diff
@@ -0,0 +1,132 @@
+		censor:   censor,
+	}
+}
+
+// client updates the censor when it interacts with secrets on the cluster
+type client struct {
+	upstream watchingclient.Client
+	censor   *secrets.DynamicCensor
+}
+
+func (c *client) Get(ctx context.Context, key ctrlruntimeclient.ObjectKey, obj ctrlruntimeclient.Object) error {
+	if err := c.upstream.Get(ctx, key, obj); err != nil {
+		return err
+	}
+	c.record(obj)
+	return nil
+}
+
+func (c *client) List(ctx context.Context, list ctrlruntimeclient.ObjectList, opts ...ctrlruntimeclient.ListOption) error {
+	if err := c.upstream.List(ctx, list, opts...); err != nil {
+		return err
+	}
+	c.recordList(list)
+	return nil
+}
+
+func (c *client) Create(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.CreateOption) error {
+	if err := c.upstream.Create(ctx, obj, opts...); err != nil {
+		return err
+	}
+	c.record(obj)
+	return nil
+}
+
+func (c *client) Delete(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.DeleteOption) error {
+	return c.upstream.Delete(ctx, obj, opts...)
+}
+
+func (c *client) Update(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.UpdateOption) error {
+	if err := c.upstream.Update(ctx, obj, opts...); err != nil {
+		return err
+	}
+	c.record(obj)
+	return nil
+}
+
+func (c *client) Patch(ctx context.Context, obj ctrlruntimeclient.Object, patch ctrlruntimeclient.Patch, opts ...ctrlruntimeclient.PatchOption) error {
+	if err := c.upstream.Patch(ctx, obj, patch, opts...); err != nil {
+		return err
+	}
+	c.record(obj)
+	return nil
+}
+
+func (c *client) DeleteAllOf(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.DeleteAllOfOption) error {
+	return c.upstream.DeleteAllOf(ctx, obj, opts...)
+}
+
+func (c *client) Watch(ctx context.Context, obj ctrlruntimeclient.ObjectList, opts ...ctrlruntimeclient.ListOption) (watch.Interface, error) {
+	return c.upstream.Watch(ctx, obj, opts...)
+}
+
+func (c *client) recordList(obj ctrlruntimeclient.ObjectList) {
+	// this code will not handle unstructured.Unstructured, but we do not
+	// think that the controller-runtime client will ever give us that
+	secretList, ok := obj.(*v1.SecretList)
+	if !ok {
+		return
+	}
+	for i := range secretList.Items {
+		c.recordSecret(&secretList.Items[i])
+	}
+}
+
+func (c *client) record(obj ctrlruntimeclient.Object) {
+	// this code will not handle unstructured.Unstructured, but we do not
+	// think that the controller-runtime client will ever give us that
+	secret, ok := obj.(*v1.Secret)
+	if !ok {
+		return
+	}
+	c.recordSecret(secret)
+}
+
+func (c *client) recordSecret(secret *v1.Secret) {
+	var values []string
+	for _, value := range secret.Data {
+		values = append(values, string(value))
+	}
+	for _, value := range secret.StringData {
+		values = append(values, value)
+	}
+	for _, key := range []string{"openshift.io/token-secret.value", "kubectl.kubernetes.io/last-applied-configuration"} {
```

Comment:
wat, how is the value of the `kubectl.kubernetes.io/last-applied-configuration` annotation secret?

Reply:
It contains the whole Secret itself, so ... :)